refinery/file_store/test_models.py

```diff
@@ -304,3 +291,60 @@ def test_translate_from_relative_path_with_no_args(self):
         translate_file_source = generate_file_source_translator()
         with self.assertRaises(ValueError):
             translate_file_source(self.rel_path_source)
+
+
+@override_storage()
+class FileImportTaskTerminationTest(TestCase):
+
+    def setUp(self):
+        self.item = FileStoreItem.objects.create()
+
+    def test_terminate_on_file_store_item_delete(self):
+        with mock.patch.object(
+                FileStoreItem, 'terminate_file_import_task'
+        ) as mock_terminate_task:
+            self.item.delete()
+            mock_terminate_task.assert_called_once()
+
+    def test_terminate_on_bulk_file_store_item_delete(self):
+        with mock.patch.object(
+                FileStoreItem, 'terminate_file_import_task'
+        ) as mock_terminate_task:
+            FileStoreItem.objects.all().delete()
+            mock_terminate_task.assert_called_once()
+
+    def test_no_terminate_on_save_with_no_new_datafile(self):
+        with mock.patch.object(
+            FileStoreItem, 'terminate_file_import_task'
+        ) as mock_terminate_task:
+            self.item.save()
+            mock_terminate_task.assert_not_called()
+
+
+@override_settings(REFINERY_DATA_IMPORT_DIR='/import/path')
```

Comment:
Does the entire test class need this decorator, or just: `test_symlink_new_item_with_import_dir_path_source`?

Reply:
I haven't checked but when settings are explicit tests are less brittle.